core/src/main/scala/com/banno/kafka/producer/ProducerOps.scala

```diff
@@ -68,13 +68,13 @@ case class ProducerOps[F[_], K, V](producer: ProducerApi[F, K, V]) {
   def transaction[G[_]: Foldable](
       records: G[ProducerRecord[K, V]],
       offsets: Map[TopicPartition, OffsetAndMetadata],
-      consumerGroupId: String,
+      groupMetadata: ConsumerGroupMetadata,
   )(implicit F: MonadError[F, Throwable]): F[Unit] =
     (for {
       _ <- producer.beginTransaction
       // should be no need to wait for RecordMetadatas or errors, since commitTransaction flushes and throws
       _ <- sendAndForgetBatch(records)
-      _ <- producer.sendOffsetsToTransaction(offsets, consumerGroupId)
+      _ <- producer.sendOffsetsToTransaction(offsets, groupMetadata)
```

Comment:
This is a change in the Java API, not a new overload?

Reply:
They deprecated the old `sendOffsetsToTransaction` and added a new one. We can do the same thing if we really need to, or just make the small breaking change.

Reply:
Ah, okay. Probably just easier to do what you did here.